modules/users/server/controllers/users/users.authentication.server.controller.js

```diff
@@ -7,48 +7,159 @@ var path = require('path'),
   errorHandler = require(path.resolve('./modules/core/server/controllers/errors.server.controller')),
   mongoose = require('mongoose'),
   passport = require('passport'),
-  User = mongoose.model('User');
+  User = mongoose.model('User'),
+  config = require(path.resolve('./config/config')),
+  nev = require('email-verification')(mongoose),
+  nodemailer = require('nodemailer');
 
 // URLs for which user can't be redirected on signin
 var noReturnUrls = [
   '/authentication/signin',
   '/authentication/signup'
 ];
 
+// NEV setup and configuration ================
+var config_nev = function () {
+
+  nev.configure({
+    tempUserCollection: "tempusers_collection",
+    verificationURL: config.host + '/api/auth/verify/${URL}',
+    persistentUserModel: User,
+
+    transportOptions: config.mailer.options,
+    verifyMailOptions: {
+      from: 'Do Not Reply <myawesomeemail_do_not_reply@gmail.com>',
+      subject: 'Please confirm account',
+      html: 'Click the following link to confirm your account:</p><p>${URL}</p>',
+      text: 'Please confirm your account by clicking the following link: ${URL}'
+    },
+    confirmMailOptions: {
+      from: config.mailer.from,
+      subject: 'Account successfully verified!',
+      html: '<p>Your account has been successfully verified.</p>',
+      text: 'Your account has been successfully verified.'
+    },
+    verifySendMailCallback: function(err, info) {
+      if (err) {
+        throw err;
+      }
+    }
+
+  }, function(err, options) {
+    if (err) {
+      throw err;
+    }
+  });
+
+  nev.generateTempUserModel(User, function(err) {
+    if (err) {
+      throw err;
+    }
+  });
+
+};
+
+config_nev();
```

Comment:
All of this "node-email-verification" functionality should be moved to it's own server controller. or server-side module (config/lib).

I think we'd be able to minimize the impact that this new feature has on the existing sign-up process.